### PR TITLE
Fixes for various module issues

### PR DESCRIPTION
**Pull request by andystaples**
- Check for host.json (#16)
- Support for relative paths (#17)
- Support for piping (#18)
- Error on function with no bindings (#20)

### CODE REVIEW

File: src/GetFunctionsMetadataCommand.cs [hunk 1]
```diff
@@ -31,25 +32,39 @@ protected override void ProcessRecord()
         {
             try
             {
-                if (!string.IsNullOrEmpty(FunctionsAppDirectory) && Directory.Exists(FunctionsAppDirectory))
+                // Attempt to normalize the user's input to a local path from where the cmdlet is running
+                SessionState ss = new SessionState();
```
Comment (Francisco-Gamino) [inline src/GetFunctionsMetadataCommand.cs:36]: Could you please add some test cases for this code path? Something like this:

```
Get-FunctionsMetadata $PSScriptRoot\TestAppOne

Get-FunctionsMetadata .\TestAppOne

Get-FunctionsMetadata .\..\TestAppOne
```

File: src/GetFunctionsMetadataCommand.cs [hunk 2]
```diff
@@ -31,25 +32,39 @@ protected override void ProcessRecord()
...
+
+                Console.WriteLine(FunctionsAppDirectoryString);
+                if (!Directory.Exists(FunctionsAppDirectoryString))
+                {
+                    ThrowTerminatingError(new ErrorRecord(new Exception(string.Format(AzPowerShellSdkStrings.InvalidFunctionsAppDirectory, FunctionsAppDirectoryString)), "InvalidFunctionAppDirectory", ErrorCategory.ParserError, null));
+                }
+
+                DirectoryInfo FunctionsAppDirectory = new DirectoryInfo(FunctionsAppDirectoryString);
+
+                if(!IsValidFunctionsAppDirectory(FunctionsAppDirectory))
```
Comment (Francisco-Gamino) [inline src/GetFunctionsMetadataCommand.cs:50]: Please add a test case for this code path as well.

File: src/GetFunctionsMetadataCommand.cs [hunk 3]
```diff
@@ -31,25 +32,39 @@ protected override void ProcessRecord()
...
+                    ThrowTerminatingError(new ErrorRecord(new Exception(string.Format(AzPowerShellSdkStrings.MissingHostJson, FunctionsAppDirectory.FullName)), "InvalidFunctionAppDirectory", ErrorCategory.ParserError, null));
+                }
+
+                List<FunctionInformation> bindingInformations = WorkerIndexingHelper.IndexFunctions(FunctionsAppDirectory, out List<ErrorRecord> errors);
+                foreach (ErrorRecord error in errors)
+                {
+                    WriteError(error);
+                }
+
+                if (PrettyPrint.IsPresent)
```
Comment (Francisco-Gamino) [inline src/GetFunctionsMetadataCommand.cs:61]: I forgot to mentioned yesterday that we should check with the team on the name of the parameters for this cmdlet. Do you mind pinging the Teams thread with a list of all customer facing parameters and ask them if they have any concerns about the parameter names? Thank you.

File: src/Microsoft.Azure.Functions.PowerShell.SDK.csproj [hunk 4]
```diff
@@ -6,12 +6,22 @@
...
+    <None Remove="test\**" />
+  </ItemGroup>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.PowerShell.SDK" Version="7.2.1" />
   </ItemGroup>
 
   <ItemGroup>
-    <Folder Include="test\E2E\apps\empty-app\" />
+    <Compile Update="resources\AzPowerShellSdkStrings.Designer.cs">
```
Comment (Francisco-Gamino) [inline src/Microsoft.Azure.Functions.PowerShell.SDK.csproj:20]: What is this file for?

File: src/WorkerIndexingHelper.cs [hunk 5]
```diff
@@ -42,25 +42,29 @@ private static void AddOutputBinding(BindingInformation outputBinding)
...
+            List<FileInfo> powerShellFiles = GetPowerShellFiles(baseDir);
             List<FunctionInformation> rpcFunctionMetadatas = new List<FunctionInformation>();
 
             foreach (FileInfo powerShellFile in powerShellFiles)
             {
                 rpcFunctionMetadatas.AddRange(IndexFunctionsInFile(powerShellFile));
             }
 
+            IEnumerable<FunctionInformation> noBindingFunctions = rpcFunctionMetadatas.Where(x => x.Bindings.Count() == 0);
+            if (noBindingFunctions.Any())
```
Comment (Francisco-Gamino) [inline src/WorkerIndexingHelper.cs:61]: If we do not have test cases for this code path, could you please add some tests? Thanks.

File: src/resources/AzPowerShellSdkStrings.Designer.cs [hunk 6]
```diff
@@ -0,0 +1,225 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//     Runtime Version:4.0.30319.42000
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
```
Comment (Francisco-Gamino) [inline src/resources/AzPowerShellSdkStrings.Designer.cs:9]: For the PowerShell language worker, we do not checkin this file. Could you please remove it and regenerate it as part of the build process?

File: src/Microsoft.Azure.Functions.PowerShell.SDK.csproj (hunk 4 above)
Comment (andystaples) [reply src/Microsoft.Azure.Functions.PowerShell.SDK.csproj:20]: Visual Studio keeps autogenerating it. I have removed it several times but it keeps coming back, so I figured I'd just leave it. It's not harmful to the resource generation

File: src/WorkerIndexingHelper.cs (hunk 5 above)
Comment (andystaples) [reply src/WorkerIndexingHelper.cs:61]: Done

File: src/resources/AzPowerShellSdkStrings.Designer.cs (hunk 6 above)
Comment (andystaples) [reply src/resources/AzPowerShellSdkStrings.Designer.cs:9]: Should I just include it in the .gitignore?

File: src/GetFunctionsMetadataCommand.cs (hunk 2 above)
Comment (andystaples) [reply src/GetFunctionsMetadataCommand.cs:50]: Done

File: src/GetFunctionsMetadataCommand.cs (hunk 1 above)
Comment (andystaples) [reply src/GetFunctionsMetadataCommand.cs:36]: Done

File: src/GetFunctionsMetadataCommand.cs (hunk 3 above)
Comment (andystaples) [reply src/GetFunctionsMetadataCommand.cs:61]: I left a comment, but if it's OK, can we let this PR go through before we get a resolution on this? The names of these parameters are not modified in this PR and I've got other changes that are blocked on this

File: src/resources/AzPowerShellSdkStrings.Designer.cs (hunk 6 above)
Comment (andystaples) [reply src/resources/AzPowerShellSdkStrings.Designer.cs:9]: I have included it in the .gitignore

File: src/GetFunctionsMetadataCommand.cs (hunk 3 above)
Comment (andystaples) [reply src/GetFunctionsMetadataCommand.cs:60]: Renamed parameter per conversation with Anatoli and Madhura

File: src/Microsoft.Azure.Functions.PowerShell.SDK.csproj (hunk 4 above)
Comment (Francisco-Gamino) [reply src/Microsoft.Azure.Functions.PowerShell.SDK.csproj:20]: Ok, so let's keep it as is for now.

**Review by Francisco-Gamino — APPROVED** [review]
LGTM
